### PR TITLE
MRG: Refactor field mapping LUT to use interp1d

### PR DESCRIPTION
**Pull request by Eric89GXL**
We had implemented nearest-neighbor and linear interpolation ourselves in the field interpolation code. This uses the more-standard `scipy.interpolate.interp1d` instead. In theory we could get some speedups, but testing shows it to be about the same. In any case, should help keep things more maintainable.

### CODE REVIEW

File: mne/forward/_lead_dots.py [hunk 1]
```diff
@@ -266,6 +229,7 @@ def _fast_sphere_dot_r0(r, rr1_orig, rr2s, lr1, lr2s, cosmags1, cosmags2s,
 
     # outer product, sum over coords
     ct = np.einsum('ik,jk->ij', rr1_orig, rr2)
+    np.clip(ct, -1, 1, ct)
```
Comment (jaeilepp) [inline mne/forward/_lead_dots.py]: Why this change?

Comment (Eric89GXL) [inline mne/forward/_lead_dots.py]: There was a value at something like `1.0000000000004` and SciPy is strict about their bounds

File: mne/forward/tests/test_field_interpolation.py [hunk 2]
```diff
@@ -196,7 +193,7 @@ def test_make_field_map_meeg():
 def _setup_args(info):
     """Helper to test_as_meg_type_evoked."""
     coils = _create_meg_coils(info['chs'], 'normal', info['dev_head_t'])
-    int_rad, noise, lut_fun, n_fact = _setup_dots('fast', coils, 'meg')
```
Comment (agramfort) [inline mne/forward/tests/test_field_interpolation.py]: why this change?

Comment (Eric89GXL) [inline mne/forward/tests/test_field_interpolation.py]: There isn't really any speed difference, so we might as well get better results. I can revert if it's a problem

**Comment by agramfort** [comment]
ok so now fast and accurate are as fast?

**Comment by Eric89GXL** [comment]
At that stage, yeah. Sometimes `fast` and `accurate` are meant to describe how many coil integration points are used. Here it's whether linear or nearest interpolation are used. I think SciPy does that interpolation in compiled code, so it seems to be a negligible difference in the context of our calculations.

**Comment by agramfort** [comment]
ok then we should just support the linear interp right? and remove the
nearest.

**Comment by Eric89GXL** [comment]
Okay with me. I'll change it now

**Comment by Eric89GXL** [comment]
Okay, now the difference between `fast` and `accurate` is just the number of coefficients used (50 vs 100). Should we get rid of `mode` altogether? That's another option.

**Comment by Eric89GXL** [comment]
... it would create deprecations in `inst.interpolate_bads`, `evoked.as_type`, and `mne.make_field_maps`

**Comment by agramfort** [comment]
how much does it impact the results?

is there any memory usage change?

**Comment by Eric89GXL** [comment]
> Okay, now the difference between fast and accurate is just the number of coefficients used (50 vs 100). 

Actually it turns out `linear` is slower than `nearest` by a factor of ~2. 50 vs 100 coefficients is also a factor of 2. So we might as well have them both.

I don't see a difference by eye between `fast` and `accurate` in `examples/preprocessing/plot_interpolate_bad_channels.py`, but who knows.

**Comment by Eric89GXL** [comment]
...reverted so that it has the same behavior as before, 50/100 coefficients and nearest/linear interp for accurate/fast respectively. Now it's back to just changing the interpolation backend.

File: mne/forward/tests/test_field_interpolation.py (hunk 2 above)
Comment (agramfort) [inline mne/forward/tests/test_field_interpolation.py]: I thought you had reverted?

Comment (Eric89GXL) [inline mne/forward/tests/test_field_interpolation.py]: It doesn't really matter here since it's a test, but I can revert it

Comment (Eric89GXL) [inline mne/forward/tests/test_field_interpolation.py]: ... although I guess it will be better for test timing to revert, too

**Comment by agramfort** [comment]
merge when CIs are happy

**Comment by Eric89GXL** [comment]
Thanks for the reviews @agramfort @jaeilepp
